client/src/App.js

```diff
@@ -109,11 +107,28 @@ class App extends Component {
 
   logState = () => console.log("App.js state finished: ", this.state);
   render() {
+    const loggedIn = this.state.logged_in
     return (
       <div className="App">
         <main>
-          <Route path="/" exact component={SplashPage}></Route>
-          <Route path="/coming_soon" exact component={ComingSoonPage}></Route>
+          {loggedIn ?
+          <Route
+            path="/"
+            exact
+            render={routerProps => (
+              <HomePage
+                handle_logout={this.handleLogout}
+                {...routerProps}
+                {...this.state}
+              />
+            )}
+          ></Route> :
+          <Route
+            path="/"
+            exact
+            component={SplashPage}
+          ></Route>
+            }
```

Comment:
This is cool! Can you just use [`<Redirect />`](https://reacttraining.com/react-router/web/api/Redirect) thought?

Reply:
Oops forgot about that part